jmh/benchmark-binary-tree/pom.xml

```diff
@@ -0,0 +1,310 @@
+                                <buildArg>--install-exit-handlers</buildArg>
+                                <buildArg>-H:+ReportExceptionStackTraces</buildArg>
+                                <buildArg>--initialize-at-build-time=org.openjdk.jmh.infra,org.openjdk.jmh.util.Utils,org.openjdk.jmh.runner.InfraControl,org.openjdk.jmh.runner.InfraControlL0,org.openjdk.jmh.runner.InfraControlL1,org.openjdk.jmh.runner.InfraControlL2,org.openjdk.jmh.runner.InfraControlL3,org.openjdk.jmh.runner.InfraControlL4</buildArg>
+                            </buildArgs>
+                       </configuration>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+        <!-- Profile to build the instrumented native binary. Run this to generate a profile -->
+        <profile>
+            <id>instrumented</id>
+            <properties>
+                <!-- Name of the output binary -->
+                <binary-name>benchmark-binary-tree-instr</binary-name>
+            </properties>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.graalvm.buildtools</groupId>
+                        <artifactId>native-maven-plugin</artifactId>
+                        <version>${native-buildtools.version}</version>
+                        <extensions>true</extensions>
+                        <executions>
+                            <!-- Run the plugin during packaging, allows for native compilation of the application -->
+                            <execution>
+                                <id>build-native</id>
+                                <phase>package</phase>
+                                <goals>
+                                    <goal>build</goal>
+                                </goals>
+                            </execution>
+                        </executions>
+                        <configuration>
+                            <!-- Set a custom name for the output binary file -->
+                            <imageName>${binary-name}</imageName>
+                            <mainClass>${main.class}</mainClass>
+                            <buildArgs>
+                                <!-- Add support for exit handlers : these are not the default. You want for
+                                 applications, such as this, but not for shared libs -->
+                                <buildArg>--install-exit-handlers</buildArg>
+                                <buildArg>-H:+ReportExceptionStackTraces</buildArg>
+                                <buildArg>--initialize-at-build-time=org.openjdk.jmh.infra,org.openjdk.jmh.util.Utils,org.openjdk.jmh.runner.InfraControl,org.openjdk.jmh.runner.InfraControlL0,org.openjdk.jmh.runner.InfraControlL1,org.openjdk.jmh.runner.InfraControlL2,org.openjdk.jmh.runner.InfraControlL3,org.openjdk.jmh.runner.InfraControlL4</buildArg>
+                                <!--buildArg>- -gc=G1</buildArg-->
```

Comment:
Is this supposed to use G1?

Reply:
Correct, the benchmark runner [uses Ubuntu](https://github.com/graalvm/graalvm-demos/pull/149/files#diff-79c638c2acbcc57a52f6b7ad32212bf3d0603af18d81c149f57f46b20ba2a19cR19) but yes, maybe we just keep this commented out. I guess the extra space between `- -` is to avoid a syntax problem.

Reply:
yes it is, commented out, but left in as only works on linux